### PR TITLE
{bio,devel} [foss/2015a] Cufflinks 2.2.1 (REVIEW)

### PR DESCRIPTION
**Pull request by rjeschmi**

### CODE REVIEW

File: easybuild/easyconfigs/c/Cufflinks/Cufflinks-2.2.1-foss-2015a.eb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+toolchainopts = {'optarch': True, 'pic': True}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = ['http://cole-trapnell-lab.github.io/cufflinks/assets/downloads/']
+
+dependencies = [
+    # issues with boost > 1.55, see https://github.com/cole-trapnell-lab/cufflinks/issues/3
+    ('Boost', '1.55.0', '-Python-2.7.9'),
+    ('SAMtools', '0.1.19'),
+    ('Eigen', '3.2.3'),
```
Comment (wpoely86) [inline easybuild/easyconfigs/c/Cufflinks/Cufflinks-2.2.1-foss-2015a.eb]: 3.2.4?

File: easybuild/easyconfigs/e/Eigen/Eigen-3.2.3-foss-2015a.eb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+version = '3.2.3'
+
+homepage = 'http://eigen.tuxfamily.org/index.php?title=Main_Page'
+description = """Eigen is a C++ template library for linear algebra:
+ matrices, vectors, numerical solvers, and related algorithms."""
+
+toolchain = {'name': 'foss', 'version': '2015a'}
+toolchainopts = {'optarch': True, 'pic': True}
+
+source_urls = ['http://bitbucket.org/%(namelower)s/%(namelower)s/get']
```
Comment (wpoely86) [inline easybuild/easyconfigs/e/Eigen/Eigen-3.2.3-foss-2015a.eb]: you can use `BITBUCKET_SOURCE` here

**Comment by wpoely86** [comment]
ok to test

File: easybuild/easyconfigs/c/Cufflinks/Cufflinks-2.2.1-foss-2015a.eb (hunk 1 above)
Comment (rjeschmi) [inline easybuild/easyconfigs/c/Cufflinks/Cufflinks-2.2.1-foss-2015a.eb]: I'm just sticking with the same version as intel-2015a, we could move them both forward if it makes sense...

Comment (wpoely86) [inline easybuild/easyconfigs/c/Cufflinks/Cufflinks-2.2.1-foss-2015a.eb]: OK, let's stick with `3.2.3`.

**Comment by rjeschmi** [comment]
Test report by @rjeschmi
**FAILED**
Build succeeded for 4 out of 39 (5 easyconfigs in this PR)
Linux centos 6.6, Intel(R) Xeon(R) CPU           E5440  @ 2.83GHz, Python 2.6.6
See https://gist.github.com/d130d423821e046cf2a9 for a full test report.

File: easybuild/easyconfigs/c/Cufflinks/Cufflinks-2.2.1-goolf-1.4.10.eb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+toolchainopts = {'optarch': True, 'pic': True}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = ['http://cole-trapnell-lab.github.io/cufflinks/assets/downloads/']
+
+dependencies = [
+    # issues with boost > 1.55, see https://github.com/cole-trapnell-lab/cufflinks/issues/3
+    ('Boost', '1.51.0', '-Python-2.7.3'),
+    ('SAMtools', '0.1.18'),
+    ('Eigen', '3.1.1'),
```
Comment (boegel) [inline easybuild/easyconfigs/c/Cufflinks/Cufflinks-2.2.1-goolf-1.4.10.eb]: these versions are quite different from the ones in the `foss/2015a` easyconfig?

Comment (rjeschmi) [inline easybuild/easyconfigs/c/Cufflinks/Cufflinks-2.2.1-goolf-1.4.10.eb]: They are consistent with the versions for goolf-1.4.10 and tied to version of samtools used in tophat (and deps there).

I had a user recently load both the foss-2015a version of cufflinks and the goolf-1.4.10 version of tophat and have trouble.

And before you ask, tophat can't be compiled against GCC-4.9 (yet?)...

Comment (boegel) [inline easybuild/easyconfigs/c/Cufflinks/Cufflinks-2.2.1-goolf-1.4.10.eb]: We usually keep dependency versions consistent across easyconfigs for the same software version (Cufflinks), across toolchains...

Comment (boegel) [inline easybuild/easyconfigs/c/Cufflinks/Cufflinks-2.2.1-goolf-1.4.10.eb]: The deps in the `foss/2015a` easyconfig contributed here match the ones in the existing one for `intel/2015a`, so just align them accordingly here too?

Comment (boegel) [inline easybuild/easyconfigs/c/Cufflinks/Cufflinks-2.2.1-goolf-1.4.10.eb]: BTW: try and keep your users from mixing toolchains, that's a big bag of worms...

Comment (rjeschmi) [inline easybuild/easyconfigs/c/Cufflinks/Cufflinks-2.2.1-goolf-1.4.10.eb]: Ok, I'll just kick the 1.4.10 version out since it isn't high priority

Comment (boegel) [inline easybuild/easyconfigs/c/Cufflinks/Cufflinks-2.2.1-goolf-1.4.10.eb]: :+1:

**Comment by rjeschmi** [comment]
removed a commit

**Comment by rjeschmi** [comment]
Test report by @rjeschmi
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux centos 6.6, Intel(R) Xeon(R) CPU           E5440  @ 2.83GHz, Python 2.6.6
See https://gist.github.com/7b7fcaea6f5ac602ebbe for a full test report.

**Comment by rjeschmi** [comment]
Test report by @rjeschmi
**SUCCESS**
Build succeeded for 9 out of 9 (4 easyconfigs in this PR)
Linux centos 6.6, Intel(R) Xeon(R) CPU           E5440  @ 2.83GHz, Python 2.6.6
See https://gist.github.com/9c8e8c0cad4bc657f6d0 for a full test report.

**Comment by rjeschmi** [comment]
The second report includes everything that is on top of the base foss-2015a install (which is why it is a bit bigger).

**Comment by rjeschmi** [comment]
@boegel I'm pretty sure all the URLs got fixed earlier this year:

https://github.com/hpcugent/easybuild-easyconfigs/pull/1335/files

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/6edca661e65692fbee3f for a full test report.

**Comment by boegel** [comment]
No reason to hold this back any further, thanks @rjeschmi (and @wpoely86 for the review)!

**Pull request merged**